### PR TITLE
Fix Bug 1183558, audit plugincheck code and add additional unit tests

### PR DESCRIPTION
**Pull request by schalkneethling**
@alexgibson r?

### CODE REVIEW

File: tests/unit/spec/plugincheck/lib/plugincheck.js [hunk 1]
```diff
@@ -220,6 +232,26 @@ describe('plugincheck.js', function() {
...
+        // if the installed version does not match any of the versions in the vulnerable array.
+        expect(plugin.status).toBe('vulnerable');
+        expect(plugin.vulnerability_description).toBe('');
+    });
+
+    it('should set plugin status to latest', function() {
+        var plugin = {};
+        var installedVersion = '13.0.0.214';
+        var plugin = PluginCheck.setPluginStatus(plugin, installedVersion, lastestRelease);
+        // as there are no latest releases, this plugin should be marked as vulnerable, even
```
Comment (alexgibson) [inline tests/unit/spec/plugincheck/lib/plugincheck.js]: Duplicate comment from the test above here?

File: tests/unit/spec/plugincheck/lib/plugincheck.js [hunk 2]
```diff
@@ -220,6 +232,26 @@ describe('plugincheck.js', function() {
...
+        var plugin = {};
+        var installedVersion = '0.0.0';
+        var plugin = PluginCheck.setPluginStatus(plugin, installedVersion, noLatestRelease);
+        // as there are no latest releases, this plugin should be marked as vulnerable, even
+        // if the installed version does not match any of the versions in the vulnerable array.
+        expect(plugin.status).toBe('vulnerable');
+        expect(plugin.vulnerability_description).toBe('');
+    });
+
+    it('should set plugin status to latest', function() {
```
Comment (alexgibson) [inline tests/unit/spec/plugincheck/lib/plugincheck.js]: Could this be a bit more descriptive?

File: tests/unit/spec/plugincheck/lib/plugincheck.js (hunk 1 above)
Comment (schalkneethling) [inline tests/unit/spec/plugincheck/lib/plugincheck.js]: meh, indeed

File: media/js/plugincheck/lib/plugincheck.js [hunk 3]
```diff
@@ -43,70 +43,68 @@
...
-        }
-    }
+            // Only test for a latest release if there are items in the array
+            if ((knownVersions.latest.length > 0) && Utils.isMatch(installedVersion, knownVersions.latest)) {
+                plugin['status'] = 'latest';
+                return plugin;
+            } else if (Utils.isMatch(installedVersion, knownVersions.vulnerable) || knownVersions.latest.length === 0) {
+                // If falling into this branch due to knownVersions.latest.length being 0,
+                // vulnerableInfo is false, leaving 'vulnerability_description' and
+                // 'vulnerability_url' as undefined. This seems to not cause any problems...
```
Comment (alexgibson) [inline media/js/plugincheck/lib/plugincheck.js]: `This seems to not cause any problems...` - this sounds a bit unsure, can we use a more descriptive explaination of what does/should happen in this case?

Comment (schalkneethling) [inline media/js/plugincheck/lib/plugincheck.js]: Sure, I can.

**Comment by alexgibson** [comment]
Other than the suggestions for making comments a bit clearer, the code changes look fine to me, and additional tests pass r+wc :beer:

**Comment by schalkneethling** [comment]
@alexgibson updated

**Comment by alexgibson** [comment]
Coveralls was stuck, but the travis CI checks all passed, so merging!

**Pull request merged**